api/agent/agent.go

```diff
@@ -691,9 +692,18 @@ func (s *hotSlot) exec(ctx context.Context, call *call) error {
 	call.req = call.req.WithContext(ctx) // TODO this is funny biz reed is bad
 
 	var errApp chan error
+	var err error
 	if call.Format == models.FormatHTTPStream {
-		errApp = s.dispatch(ctx, call)
+		req, err := callToHTTPRequest(ctx, call)
+		// the acksync channel is used on on LB configuration so the channel could be not initialized
+		if call.ackSync != nil {
+			call.ackSync <- err
```

Comment:
is there a reason to plumb this here as opposed to `call.Start()` ? semantically, callToHTTPRequest itself cannot fail unless we have a programming error, this seems like something we should fix up in general, which seems like it would be the motivation for putting this down here instead of `call.Start()` since there's no other error between here and there? at least, call.Start() and call.End() were purpose built for async processing, we should either repurpose them to our needs (maybe with some maintenance/changes) or get rid of them and plumb everything in directly (as is the case here).

Reply:
(at least, callToHTTPRequest is not part of the critical path and should be encapsulated much better / we shouldn't be plumbing around it)

Reply:
did the honors... https://github.com/fnproject/fn/pull/1264 sorry for the rebase, hopefully helpful...

Reply:
With your change to the callToHTTPRequest I think there is no reasons now to put the ack down to the exec call.